crates/fuzzing/src/generators/codegen_settings.rs

```diff
@@ -32,6 +33,46 @@ impl CodegenSettings {
             }
         }
     }
+
+    /// Features such as sse4.2 are unconditionally enabled on the x86_64 target
+    /// because they are hard required for SIMD, but when SIMD is disabled, for
+    /// example, we support disabling these features.
+    ///
+    /// This method will take the wasm feature selection chosen, through
+    /// `module_config`, and possibly try to disable some more features by
+    /// reading more of the input.
+    pub fn maybe_disable_more_features(
+        &mut self,
+        module_config: &ModuleConfig,
+        u: &mut Unstructured<'_>,
+    ) -> arbitrary::Result<()> {
+        let flags = match self {
+            CodegenSettings::Target { flags, .. } => flags,
+            _ => return Ok(()),
+        };
+
+        if !module_config.config.simd_enabled {
+            // Note that regardless of architecture these booleans are generated
+            // to have test case failures unrelated to codegen setting input
+            // that fail on one architecture to fail on other architectures as
+            // well.
+            let sse3 = u.arbitrary::<bool>()?;
+            let ssse3 = u.arbitrary::<bool>()?;
+            let sse4_1 = u.arbitrary::<bool>()?;
+            let sse4_2 = u.arbitrary::<bool>()?;
+
+            for (name, val) in flags {
+                match name.as_str() {
+                    "has_sse3" => *val = sse3.to_string(),
+                    "has_ssse3" => *val = ssse3.to_string(),
+                    "has_sse41" => *val = sse4_1.to_string(),
+                    "has_sse42" => *val = sse4_2.to_string(),
+                    _ => {}
+                }
+            }
```

Comment:
> Is it okay to turn some off without turning off later ones?

While I'm not 100% certain myself I believe that this question is up to Cranelift mostly. I believe Cranelift is mostly structured around "if this feature is active emit this instr" while it doesn't make sense to disable sse3 but leave sse4.1 enabled I don't think it will break anything since it would be a sort of "chaos mode" for cranelift stressing it.

If this becomes a problem for Cranelift, however, we can tweak the fuzz input to respect what the actual CPU feature hierarchy is.